### PR TITLE
Support RENAME TABLE statement

### LINKED ISSUE #130: Support RENAME TABLE statement
`RENAME TABLE` statement is not yet supported.

https://cloud.google.com/spanner/docs/reference/standard-sql/data-definition-language#rename_table

```
RENAME TABLE old_table_name TO new_table_name ...
   [, old_table_name2 TO new_table_name2 ...]
```

Related

-  #115

### PR DESCRIPTION
**Pull request by apstndb**
This PR implements `RENAME TABLE` statement.

fix #130

### CODE REVIEW

File: ast/ast.go [hunk 1]
```diff
@@ -2051,6 +2053,29 @@ type DropTable struct {
...
+
+	Rename token.Pos // position of "RENAME" pseudo keyword
+
+	SrcDests []*RenameTableSrcDest // len(SrcDests) > 0
+}
+
+// RenameTableSrcDest is old_table_name TO new_table_name node in RENAME TABLE statement.
+//
+//	{{.Src | sql}} TO {{.Dest | sql}}
+type RenameTableSrcDest struct {
```
Comment (makenowjust) [inline ast/ast.go:2071]: `RenameTableTo` sounds like a better name to me than this. At least, it seems that "src" and "dest" do not appear in the document.

Comment (apstndb) [reply ast/ast.go]: It is reasonable comment and LTGM `RenameTableTo `.
Fixed.

File: ast/ast.go [hunk 2]
```diff
@@ -2051,6 +2053,29 @@ type DropTable struct {
...
+
+// RenameTableTo is src TO dest node in RENAME TABLE statement.
+//
+//	{{.Src | sql}} TO {{.Dest | sql}}
+type RenameTableTo struct {
+	// pos = Src.pos
+	// end = Dest.end
+
+	Src  *Ident
+	Dest *Ident
```
Comment (makenowjust) [inline ast/ast.go:2076]: Why are you persistent with `src` and `dest` naming? The document says `old` and `new`, so we can follow it.

File: ast/ast.go [hunk 3]
```diff
@@ -2051,6 +2053,29 @@ type DropTable struct {
...
+// RenameTable is RENAME TABLE statement node.
+//
+//	RENAME TABLE {{.ToList | sqlJoin ", "}}
+type RenameTable struct {
+	// pos = Drop
+	// end = Name.end
+
+	Rename token.Pos // position of "RENAME" pseudo keyword
+
+	ToList []*RenameTableTo // len(ToList) > 0
```
Comment (makenowjust) [inline ast/ast.go:2065]: I think `Tos` is not a problem in this case.

**Review by makenowjust — CHANGES_REQUESTED** [review]

File: ast/ast.go (hunk 2 above)
Comment (apstndb) [reply ast/ast.go]: Thank you. Fixed.

File: ast/ast.go (hunk 3 above)
Comment (apstndb) [reply ast/ast.go]: Thank you.
Fixed. https://github.com/cloudspannerecosystem/memefish/pull/170/commits/527fb7b0a8702b62af4996c469ee3a0f3e60b460

File: ast/ast.go [hunk 4]
```diff
@@ -2051,6 +2053,29 @@ type DropTable struct {
 	Name     *Ident
 }
 
+// RenameTable is RENAME TABLE statement node.
+//
+//	RENAME TABLE {{.Tos | sqlJoin ", "}}
+type RenameTable struct {
+	// pos = Drop
```
Comment (makenowjust) [inline ast/ast.go:2060]:
```suggestion
	// pos = Rename
```

File: ast/ast.go [hunk 5]
```diff
@@ -2051,6 +2053,29 @@ type DropTable struct {
...
+	// end = Name.end
```
Comment (makenowjust) [inline ast/ast.go:2061]:
```suggestion
	// end = Tos[$].end
```

Comment (apstndb) [reply ast/ast.go]: Fixed https://github.com/cloudspannerecosystem/memefish/pull/170/commits/6969f0447cbf2c9483d18224848e5b2e587eafac

File: ast/ast.go (hunk 4 above)
Comment (apstndb) [reply ast/ast.go]: Fixed https://github.com/cloudspannerecosystem/memefish/pull/170/commits/6969f0447cbf2c9483d18224848e5b2e587eafac

**Pull request merged**